src/core/net/dns_client.hpp

```diff
@@ -724,6 +831,33 @@ class Client : public InstanceLocator, private NonCopyable
 
 #endif // OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
 
+#if OPENTHREAD_CONFIG_DNS_CLIENT_ARBITRARY_RECORD_QUERY_ENABLE
+    /**
+     * Sends a DNS query for a given record type and name.
+     *
+     * The @p aConfig can be `nullptr`. In this case the default config (from `GetDefaultConfig()`) will be used as
+     * the config for this query. In a non-NULL @p aConfig, some of the fields can be left unspecified (value zero).
```

Comment:
It says "in a non-NULL aConfig" - but if aConfig is NULL, then the fields left unspecified will be taken from the default config, right?  So it doesn't matter if aConfig is NULL or not?

Reply:
Yes, exactly. If it is `NULL`, everything (in the config) is effectively unspecified, and therefore the specified default configuration values will be used.

The description elaborates a bit more on the `NULL` case to make it clear that the full `GetDefaultConfig()` values will be used.